enhancements/proposals/2457-nodeconfig-sysctls/README.md

```diff
@@ -0,0 +1,124 @@
+
+```golang
+type NodeConfigSpec struct {
+    ...
+	
+    // sysctls specifies a list of sysctls to configure on the node.
+    // +optional
+    Sysctls []corev1.Sysctl `json:"sysctls,omitempty"`
+}
+```
+
+#### `.spec.disableOptmizations` and `.spec.sysctls`
+
+The existing `.spec.disableOptimizations` field will not affect the configuration of specified sysctls.
+The sysctls specified in `.spec.sysctls` will always be configured, regardless of the value of `.spec.disableOptimizations`.
+To minimize the risk of confusion, the `.spec.disableOptmizations` comment will be adjusted to clarify that it only affects performance tuning.
+
+To disable the configuration of sysctls, the user should not specify the `.spec.sysctls` field.
+However, removing values from `.spec.sysctls` will not revert the earlier configured sysctls to their initial values.
+
+#### NodeConfig Status
+
+The semantics of the existing NodeConfig status' `TunedNode` field will be extended to also take into account whether the specified sysctls were successfully configured on all matching nodes.
+
+### NodeTune controller
+
+The existing NodeTune controller, running as part of the node setup DaemonSet, will be extended to handle the sysctls configuration.
+The controller will create dedicated Jobs configuring sysctls on the nodes matching the placement rules specified in the NodeConfig.
+
+### ScyllaDB recommended sysctls
+
+ScyllaDB Operator will not provide configure any sysctls by default, excluding any sysctls independently configured by the perftune script.
+Users are expected to configure sysctls based on their specific use cases and requirements, including the number of ScyllaDB nodes running on Kubernetes nodes matched by NodeConfig's placement rules.
+However, the ScyllaDB Operator documentation will provide the recommended sysctls for nodes running ScyllaDB Pods, as specified in the ScyllaDB's GitHub repository: https://github.com/scylladb/scylladb/tree/7bb43d812e1c512c92541a93603fcea0499e6d05/dist/common/sysctl.d.
+In particular, the documentation will highlight which sysctls should be set to specific values and which should adjusted to the number of ScyllaDB nodes running on the same Kubernetes node.
+There is no reasonable way to automate the suggested configuration, so the documentation will be updated on a best-effort basis to reflect the latest recommendations from ScyllaDB repository.
+
+### `ScyllaCluster.scylla.scylladb.com/v1.spec.sysctls` deprecation
+
+The existing sysctl configuration option in `ScyllaCluster.scylla.scylladb.com/v1` will be marked as deprecated.
+Users will be advised to remove the deprecated `.spec.sysctls` field from their `ScyllaCluster.scylla.scylladb.com/v1` resources before using the new `NodeConfig.scylla.scylladb.com/v1alpha1.spec.sysctls`, as otherwise the two might result in performing conflicting actions.
+
+### Test Plan
+
+Validation of the sysctl configuration and controller logic will be covered by unit tests.
```

Comment:
Shouldn't we have E2E coverage to ensure the sysctls indeed get correctly set?

Reply:
updated